telemetry/prometheus/metrics.go

```diff
@@ -196,14 +198,15 @@ func (p *metrics) Time(name string, value time.Duration, tags []string) {
 			Namespace: p.cfg.Namespace,
 			Subsystem: p.cfg.Subsystem,
 			Help:      name + " timing histogram",
-			Buckets:   prometheus.LinearBuckets(0, 1, 10), // Adjust bucketing as necessary
+			// Given bucket=0.01s(10ms), the maximum covered time range is 10ms * TimeBucketCount
+			Buckets: prometheus.LinearBuckets(0, 0.01, p.cfg.TimeBucketCount),
 		}, labels)
 		prometheus.MustRegister(histogramVec)
 		p.histogramVecs[name] = histogramVec
 	}
 
 	// Convert time.Duration to seconds since Prometheus prefers base units
-	histogramVec.WithLabelValues(labelValues...).Observe(value.Seconds())
+	histogramVec.WithLabelValues(labelValues...).Observe(float64(value.Seconds()))
```

Comment:
linter says the float64 cast is uneccesary here

Reply:
Removed. Good catch!